sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -94,6 +94,10 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     try {
       body
     } catch {
+      case i: InvocationTargetException if isClientException(i.getTargetException) =>
+        val e = i.getTargetException
+        throw new AnalysisException(
+          e.getClass.getCanonicalName + ": " + e.getMessage, cause = Some(e))
```

Comment:
This fix needs to be backported to the previous releases. Will submit a separate one for this.